### PR TITLE
Ref-ESSNTL-3394 Run unit tests in a container

### PR DESCRIPTION
**Pull request by r14chandra**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,5 +6,11 @@ RUN microdnf install -y python38 python38-devel curl python3-pip git tar xz bzip
 COPY src src
 COPY poetry.lock poetry.lock
 COPY pyproject.toml pyproject.toml
+COPY unit_test.sh unit_test.sh
+COPY requirements.txt requirements.txt
+COPY dev dev
+COPY tests tests
+COPY cdappconfig.json cdappconfig.json
+COPY parse_json.py parse_json.py
```
Comment (patilsuraj767) [inline Dockerfile:14]: Can we shift `cdappconfig.json` and `parse_json.py` under dev directory as these files are not related to the production code. Doing this will reduce Dockerfile lines and image layers while building the image. 
Also please shift line 6 `COPY src src` after `RUN pip3 install` this will reduce the image build time (while building docker will use old unchanged image layers).

File: unit_test.sh [hunk 2]
```diff
@@ -1,37 +1,41 @@
...
-     python inventory-schemas/tools/simple-test/tester.py inventory-schemas/schemas/system_profile/v1.yaml output.json
-     if [ $? != 0 ]; then
-        exit 1
-     fi
-     rm $filename
-     rm -rf inventory-schemas
-     rm output.json
+   filename="$(basename "$file").tar.gz"
+   tar -zcf $filename "$file"
+   insights-run -p src/puptoo -f json $filename > output.json
```
Comment (patilsuraj767) [inline unit_test.sh:26]:
```suggestion
   insights-run -p src/puptoo -f json $filename > dev/output.json
```
As we are shifting `parse_json.py` inside `dev` this will make sure script does not break.

File: unit_test.sh [hunk 3]
```diff
@@ -1,37 +1,41 @@
...
+   insights-run -p src/puptoo $filename # this is to print the results in the test console
+   if [ $? != 0 ]; then
+      exit 1
+   fi
+   python parse_json.py
```
Comment (patilsuraj767) [inline unit_test.sh:31]:
```suggestion
   python dev/parse_json.py
```

**Review by patilsuraj767 — CHANGES_REQUESTED** [review]
@r14chandra Overall PR looks good to me. Added few suggestions just to make code/repository look cleaner.

File: Dockerfile (hunk 1 above)
Comment (r14chandra) [reply Dockerfile]: Thanks for the review! Refactored the code.

File: Dockerfile [hunk 4]
```diff
@@ -1,10 +1,19 @@
...
 COPY pyproject.toml pyproject.toml
-RUN pip3 install --upgrade pip && pip3 install .  
-CMD ["puptoo"]
\ No newline at end of file
+COPY unit_test.sh unit_test.sh
+COPY requirements.txt requirements.txt
+COPY dev dev
+COPY tests tests
+
+CMD ["puptoo"]
```
Comment (patilsuraj767) [inline Dockerfile:19]: File should look like below. 
```
FROM registry.redhat.io/ubi8/ubi-minimal

WORKDIR /app-root/

RUN microdnf install -y python38 python38-devel curl python3-pip git tar xz bzip2 unzip && \
    git clone -b 3.0 https://github.com/RedHatInsights/insights-core && \
    pip3 install ./insights-core


COPY poetry.lock poetry.lock
COPY pyproject.toml pyproject.toml
RUN pip3 install --upgrade pip && pip3 install . 
COPY unit_test.sh unit_test.sh
COPY requirements.txt requirements.txt
COPY dev dev
COPY tests tests 
COPY src src
CMD ["puptoo"]
```

**Review by patilsuraj767 — APPROVED** [review]
LGTM :+1:
